r2i/edgetpu/engine.cc

```diff
@@ -0,0 +1,98 @@
+/* Copyright (C) 2018-2020 RidgeRun, LLC (http://www.ridgerun.com)
+ * All Rights Reserved.
+ *
+ * The contents of this software are proprietary and confidential to RidgeRun,
+ * LLC.  No part of this program may be photocopied, reproduced or translated
+ * into another programming language without prior written consent of
+ * RidgeRun, LLC.  The user is free to modify the source code after obtaining
+ * a software license from RidgeRun.  All source code changes must be provided
+ * back to RidgeRun without any encumbrance.
+ */
+
+#include "r2i/edgetpu/engine.h"
+
+#include <edgetpu.h>
+
+namespace r2i {
+namespace edgetpu {
+
+Engine::Engine () {
+  this->state = r2i::tflite::Engine::State::STOPPED;
+  this->model = nullptr;
+  this->number_of_threads = 1;
+  this->allow_fp16 = 0;
+}
+
+Engine::~Engine () {
+  r2i::tflite::Engine::Stop();
+}
+
+void Engine::setupResolver(::tflite::ops::builtin::BuiltinOpResolver
+                           &resolver) {
+  resolver.AddCustom(::edgetpu::kCustomOp, ::edgetpu::RegisterCustomOp());
+}
+
+void Engine::setInterpreterContext() {
+  std::shared_ptr<::edgetpu::EdgeTpuContext> edgetpu_context =
+    ::edgetpu::EdgeTpuManager::GetSingleton()->OpenDevice();
+  this->interpreter->SetExternalContext(kTfLiteEdgeTpuContext,
+                                        edgetpu_context.get());
+}
+
+float *Engine::runInference(std::shared_ptr<r2i::IFrame> frame,
+                            const int &input, const int &width, const int &height, const int &channels,
+                            r2i::RuntimeError &error) {
+  auto input_tensor = this->interpreter->typed_tensor<uint8_t>(input);
+  auto input_data = (float *)frame->GetData();
+
+  if (!input_data) {
```

Comment:
Plenty this code is replicated with the parent's `RunInference`. Should we segregate it even more? Something like: `PrepareInput` and `PrepareOutput` or something like that.